packages/eds-core-react/src/components/Switch/Switch.stories.tsx

```diff
@@ -1,67 +1,65 @@
-import { useState, useEffect } from 'react'
+import { useState, useEffect, ChangeEvent } from 'react'
 import { Switch, SwitchProps, EdsProvider, Table, Density } from '../..'
 import styled from 'styled-components'
-import { Meta, Story } from '@storybook/react'
+import { ComponentMeta, Story } from '@storybook/react'
 import { data } from '../../stories/data'
+import page from './Switch.docs.mdx'
 
-type WrapperStyleProps = {
-  darkMode?: boolean
-}
-const Wrapper = styled.div<WrapperStyleProps>`
-  display: grid;
-  grid-template-rows: min-width;
-  padding: 32px;
-  padding-bottom: 8rem;
-  grid-gap: 2rem;
-  position: relative;
-  background-color: ${({ darkMode }) => (darkMode ? '#0A0310' : 'white')};
-  color: ${({ darkMode }) => (darkMode ? 'white' : 'black')};
-  transition: all 0.36s;
+const UnstyledList = styled.ul`
+  margin: 0;
+  padding: 0;
+  list-style-type: none;
 `
 
 export default {
   title: 'Inputs/Selection Controls/Switch',
   component: Switch,
   parameters: {
     docs: {
-      description: {
-        component: `Selection controls allow users to select options, make decisions and set preferences.
-        `,
-      },
+      page,
     },
   },
-} as Meta
+} as ComponentMeta<typeof Switch>
 
-export const Default: Story<SwitchProps> = (args) => (
+export const Introduction: Story<SwitchProps> = (args) => (
   <Switch label="Play with me" {...args} />
 )
 
 export const DefaultStates: Story<SwitchProps> = () => {
-  const UnstyledList = styled.ul`
-    list-style-type: none;
-    li {
-      margin-bottom: 8px;
-    }
-  `
+  const [check, setCheck] = useState(false)
+
   return (
-    <Wrapper>
-      <UnstyledList>
-        <li>
-          <Switch label="I'm default off" />
-        </li>
-        <li>
-          <Switch label="I'm default on" defaultChecked />
-        </li>
-        <li>
-          <Switch disabled label="You can't turn me on!" />
-        </li>
-        <li>
-          <Switch disabled defaultChecked label="You can't turn me off!" />
-        </li>
-      </UnstyledList>
-    </Wrapper>
+    <UnstyledList>
+      <li>
+        <Switch label="I'm default off" />
+      </li>
+      <li>
+        <Switch label="I'm default on" defaultChecked />
+      </li>
+      <li>
+        <Switch disabled label="You can't turn me on!" />
+      </li>
+      <li>
+        <Switch disabled defaultChecked label="You can't turn me off!" />
+      </li>
+      <li>
+        <Switch
+          label="I'm a controlled component"
```

Comment:
Could perhaps the label be updated to say "I'm a controlled label: checked" when true and unchecked when false (or some other visual change)

Reply:
Updated

Reply:
Getting closer, but you can simplify this a bit more by removing that outside label and do this instead, then you also don't need the extra useState:
`label={\`Output from slider is ${check ? 'checked' : 'unchecked'}\`}`

Reply:
Done